Snowflake.Data/Core/Converter/JsonToStructuredTypeConverter.cs

```diff
@@ -156,14 +156,22 @@ private static object ConvertToUnstructuredType(FieldMetadata fieldMetadata, Typ
             {
                 var value = json.Value<string>();
                 var bytes = Encoding.UTF8.GetBytes(value);
+                if ((value.Contains("e") || value.Contains("E")) && double.TryParse(value, out double doubleValue))
```

Comment:
I'm thinking if in case of float `double.TryParse(value, out double doubleValue)` and then casting `(float)doubleValue` is always the same as `float.TryParse(value, out float floatValue)`?
Because we could alternatively check fieldType and parse as float or double without casting.

Reply:
The values are equal on the test

Switched to parsing based on the `fieldType` instead of casting